dialogue-core/src/main/java/com/palantir/dialogue/core/DialogueChannel.java

```diff
@@ -242,8 +250,15 @@ private static ImmutableList<LimitedChannel> createHostChannels(
                 channel = HostMetricsChannel.create(cf, channel, targetUri.uri());
                 channel =
                         new TraceEnrichingChannel(channel, DialogueTracing.tracingTags(cf, uriIndexForInstrumentation));
-                channel = new DeadlineAdvertisementChannel(
-                        channel, cf.clientConf().readTimeout());
+                if (cf.deadlineEnforcementEnabled().isPresent()) {
```

Comment:
rethinking this a bit, i wonder if the `Optional<Boolean>` is really necessary here? we have a ctor overload for DeadlineAdvertisementChannel such that passing a boolean either enables/disables enforcement, and the alternative just defaults to DEFER, so perhaps the Optional isn't necessary to expose to callers?